grc.fraudDetection_web/src/main/java/com/typedb/examples/fraud/web/GqlResource.java

```diff
@@ -87,4 +87,11 @@ public Set<Transaction> getSuspectTransactions() {
 
     return transactions.getSuspect();
   }
+
+  @Query
+  @Description("Get all transactions with limit and offset")
+  public Set<Transaction> getTransactionsLimitOffset(@Name("limit") int limit, @Name("offset") int offset){
+
+    return transactions.getLimitOffset(limit, offset);
+  }
 }
```

Comment:
Optionally, we could just add the parameters to the existing method if we check for null deeper in the DAO.